"What's new in Svelte" December newsletter (#5706)
sveltejs/svelte: site/content/blog/2020-12-01-whats-new-in-svelte-december-2020.md

---
title: What's new in Svelte: December 2020
description: TBD
author: Daniel Sandoval
authorURL: https://desandoval.net
---

> It's the last "What's new in Svelte" of the year... To round out the season, this month's coverage includes _things_!

Reviewer: What is meant by _things_? I would prefer a more specific wording, like "a lot of showcases".

Author: Sorry this is a placeholder for when it's closer to the end of the month. The "TBD" in the description is also a placeholder. :)

> ## New features & impactful bug fixes

Reviewer 2: it's not released yet but https://github.com/sveltejs/svelte/pull/5584 is probably the biggest feature. sourcemaps for preprocessor means that sourcemaps will now work with typescript and other languages

Reviewer 2: @d3sandoval do we want to add something about #5584?

Author: Doesn't number 5 on my list cover that?

Reviewer 2: Number 5 looks like it's about export maps, which doesn't sound related at all.

#5584 let you use source maps with TypeScript, Sass, Less, etc. in Svelte

Reviewer: I just tested with the latest release on the svelte template converted to TS and I still have this offset error. I think it's a huge feature but I feel it's not quite ready yet and needs some more polish. I suggest to move the official announcement to the next monthly edition.

Reviewer 2: Ah, right. I forgot about the offset error (https://github.com/sveltejs/svelte/pull/5584#issuecomment-733811305). @milahu had been looking at that one too and had some ideas, but it probably won't be fixed before the newsletter goes out, so saving for next month sounds reasonable.

> ## New features & impactful bug fixes

Reviewer 3: up to u to decide whether to include them

> ## New features & impactful bug fixes

Author: added in latest commit: https://github.com/d3sandoval/svelte/commit/77fcc97acbe5f26ef227691d8fd264a09e7fe252

1. Transitions within `{:else}` block should now complete successfully (**3.29.5**, [Example](https://svelte.dev/repl/49cef205e5da459594ef2eafcbd41593?version=3.29.5))
2. Svelte now includes an export map, which explicitly states which files can be imported from its npm package (**3.29.5** with some fixes in **3.29.6** and **3.29.7**)

> 

Reviewer 2: `rollup-plugin-svelte` had a [new 7.0.0 release](https://github.com/sveltejs/rollup-plugin-svelte/blob/master/CHANGELOG.md). Biggest change is that the `css` option was removed. Users who were using that option should add another plugin like `rollup-plugin-css-only` [as demonstrated in the template](https://github.com/sveltejs/template/blob/5b1135c286f7a649daa99825a077586655051649/rollup.config.js#L48)

## What's going on in Sapper?

> Lots of new typecript definitions to make editing Sapper apps even easier! Dynamic imports also should now work in `client.js` files. (Unreleased)

Reviewer 2: ```suggestion
Lots of new TypeScript definition improvements to make editing Sapper apps even easier! CSS for dynamic imports also should now work in `client.js` files. (Unreleased)
```

## What's the deal with SvelteKit?
We're glad you asked! If you didn't catch Rich's blog post from early last month, [you can find it here](https://svelte.dev/blog/whats-the-deal-with-sveltekit)!

For all the features and bugfixes see the CHANGELOGs for [Svelte](https://github.com/sveltejs/svelte/blob/master/CHANGELOG.md) and [Sapper](https://github.com/sveltejs/sapper/blob/master/CHANGELOG.md).

---

## Community Showcase

> **Apps & Sites**

Reviewer 4: Not sure if it fits here, but https://www.johnells.se was a big release on Svelte/Sapper which utilises Findify, Sanity.io, and Jetshop, and is getting a bit of press. Full disclosure: I worked on this, please ignore this if it's not of interest! It has already been shared via twitter @ https://twitter.com/SvelteSociety/status/1321811444606722050

---
title: What's new in Svelte: December 2020
description: Better tooling, export maps and improvements to slots and context
author: Daniel Sandoval
authorURL: https://desandoval.net
---

It's the last "What's new in Svelte" of the year and there's lots to celebrate! This month's coverage includes updates from `rollup-plugin-svelte`, `Sapper` and `SvelteKit` and a bunch of showcases from the Svelte community!

## New features & impactful bug fixes

1. `slot` components now support spread props: `<slot {...foo} />` (**3.30.0**)
2. A new `hasContext` lifecycle function makes it easy to check whether a `key` has been set in the context of a parent component (**3.30.0**, [Docs](https://svelte.dev/docs#hasContext))

> 3. A new Typed `SvelteComponent` makes it easier to add typed classes that extends base Svelte Components. Component library and framework authors rejoice! (**3.30.0**, [RFC](https://github.com/sveltejs/rfcs/pull/37))

Reviewer: ```suggestion
3. A new Typed `SvelteComponent` makes it easier to add typed classes that extends base Svelte Components. Component library and framework authors rejoice! An example: `export class YourComponent extends SvelteComponent<{aProp: boolean}, {click: MouseEvent}, {default: {aSlot: string}}> {}` (**3.30.0**, [RFC](https://github.com/sveltejs/rfcs/pull/37))
```

4. Transitions within `{:else}` blocks should now complete successfully (**3.29.5**, [Example](https://svelte.dev/repl/49cef205e5da459594ef2eafcbd41593?version=3.29.5))

> 5. Svelte now includes an export map, which explicitly states which files can be imported from its npm package. This should significantly improve debugging in web dev tools (**3.29.5** with some fixes in **3.29.6**, **3.29.7** and **3.30.0**)

Reviewer 2: > This should significantly improve debugging in web dev tools

oh cool. I didn't know that. It might be interesting to link this section of text to an article discussing that if one exists

Reviewer: I also did not know that / cannot find anything about better debugging in that context. It helps to know what files to retrieve from which paths, and to close off private paths, but I did not find anything about debugging.

```suggestion
5. Svelte now includes an export map, which explicitly states which files can be imported from its npm package (**3.29.5** with some fixes in **3.29.6**, **3.29.7** and **3.30.0**)
```

> 6. `rollup-plugin-svelte` had a new [7.0.0 release](https://github.com/sveltejs/rollup-plugin-svelte/blob/master/CHANGELOG.md). Biggest change is that the css option was removed. Users who were using that option should add another plugin like `rollup-plugin-css-only` as demonstrated [in the template](https://github.com/sveltejs/template/blob/5b1135c286f7a649daa99825a077586655051649/rollup.config.js#L48)

Reviewer 2: ```suggestion
6. `rollup-plugin-svelte` had a new [7.0.0 release](https://github.com/sveltejs/rollup-plugin-svelte/blob/master/CHANGELOG.md). The biggest change is that the `css` option was removed. Users who were using that option should add another plugin like `rollup-plugin-css-only` as demonstrated [in the template](https://github.com/sveltejs/template/blob/5b1135c286f7a649daa99825a077586655051649/rollup.config.js#L48)
```
